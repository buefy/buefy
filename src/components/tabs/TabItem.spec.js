import { shallowMount } from '@vue/test-utils'
import BTabItem from '@components/tabs/TabItem'

let wrapper
const BTabs = {
    template: '<b-tab-stub></b-tab-stub>',
    data() {
        return {
            tabItems: [],
            _isTabs: true
        }
    }
}

describe('BTabItem', () => {
    beforeEach(() => {
        wrapper = shallowMount(BTabItem, {
            parentComponent: BTabs
        })
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('BTabItem')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('set isActive when activate is called', () => {
        wrapper.vm.activate()
        expect(wrapper.vm.isActive).toBeTruthy()
    })

    it('reset isActive when deactivate is called', () => {
        wrapper.vm.deactivate()
        expect(wrapper.vm.isActive).toBeFalsy()
    })
})
