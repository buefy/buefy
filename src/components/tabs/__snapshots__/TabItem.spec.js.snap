// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BTabItem render correctly 1`] = `<div class="tab-item" style="display: none;"></div>`;
